src/org/mozilla/javascript/NativeJSON.java

```diff
@@ -500,6 +542,27 @@ private static String quote(String string) {
         return product.toString();
     }
 
+    private static Object javaToJSON(Object value, StringifyState state) {
+        value = state.cx.getJavaToJSONConverter().apply(value);
+        value = Context.javaToJS(value, state.scope, state.cx);
+        return stringify(state.cx, state.scope, value, state.replacer, state.gap);
+    }
+
+    private static boolean isComplexJavaObject(Object o) {
```

Comment:
I use e4x/xml quite a bit, and I took the naming from there. An element has complex content if it has child elements, and simple content if it does not.

Would `isJavaContainer` or something like that be more clear? I was trying to indicate that it would not be converted  as a single java Object and instead processed recursively by `ja` or `jo`. I couldn't think of a good term for that. `isJavaArrayOrCollectionOrMap` or `isConvertableToNativeObjectOrNativeArray` seemed too verbose, but maybe one of those are clearer?

A `String[]` is still an instance of `Object[]`, but I did find there is an issue with arrays of primitive types (like `byte[]`) which should be corrected by using `o.getClass().isArray()`, like you suggested.

Reply:
I could also inline the check, but I thought I was improving readability by breaking it out to the function :shrug:  :smiley:

Reply:
I removed this method entirely, as it was only called once.